projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.integration.spec.ts

```diff
@@ -228,173 +210,81 @@ describe('IgxHierarchicalGrid Integration #hGrid', () => {
-            // expand 1st row
-            (firstRow.nativeElement.children[0] as HTMLElement).click();
+        it('should display correct child data for expanded row after sorting.', fakeAsync(() => {
+            /* this test doesn't need scrolling as it only cares about the child grid getting assigned to the correct parent */
+            hierarchicalGrid.data = hierarchicalGrid.data.slice(0, 3);
             fixture.detectChanges();
+            // expand first row
+            hierarchicalGrid.expandRow(hierarchicalGrid.dataRowList.first.rowID);
             hierarchicalGrid.sort({
                 fieldName: 'ID', dir: SortingDirection.Desc, ignoreCase: false, strategy: DefaultSortingStrategy.instance()
             });
             fixture.detectChanges();
-
-            hierarchicalGrid.verticalScrollContainer.scrollTo(hierarchicalGrid.dataView.length - 1);
-            await wait(100);
-            fixture.detectChanges();
-            hierarchicalGrid.verticalScrollContainer.scrollTo(hierarchicalGrid.dataView.length - 1);
-            await wait(100);
-            fixture.detectChanges();
-
-            const childGrid =  hierarchicalGrid.hgridAPI.getChildGrids(false)[0];
-            const fChildCell =  childGrid.dataRowList.toArray()[0].cells.toArray()[0];
+            const childGrid = hierarchicalGrid.hgridAPI.getChildGrids(false)[0];
+            const firstChildCell = childGrid.dataRowList.first.cells.first;
+            expect(hierarchicalGrid.getRowByIndex(3) instanceof IgxChildGridRowComponent).toBeTruthy();
             expect(childGrid.data).toBe(fixture.componentInstance.data[0]['childData']);
-            expect(fChildCell.value).toBe('00');
+            expect(firstChildCell.value).toBe('00');
         }));
 
-        it('should allow sorting via headers in child grids', fakeAsync(/** row toggle rAF */() => {
-            const firstRow = hierarchicalGrid.dataRowList.toArray()[0];
-            // expand 1st row
-            (firstRow.nativeElement.children[0] as HTMLElement).click();
-            fixture.detectChanges();
+        it('should allow sorting via headers in child grids', fakeAsync(() => {
+            // expand first row
+            hierarchicalGrid.expandRow(hierarchicalGrid.dataRowList.first.rowID);
             // enable sorting
-            const childGrid =  hierarchicalGrid.hgridAPI.getChildGrids(false)[0];
-            childGrid.columnList.toArray()[0].sortable = true;
+            const childGrid = hierarchicalGrid.hgridAPI.getChildGrids(false)[0];
+            childGrid.columnList.first.sortable = true;
             fixture.detectChanges();
 
-            const childHeaders = fixture.debugElement.query(By.css('igx-child-grid-row')).queryAll(By.css('igx-grid-header'));
-            childHeaders[0].nativeElement.click();
+            const childHeader = fixture.debugElement.query(By.css('igx-child-grid-row')).queryAll(By.css('igx-grid-header'))[0];
```

Comment:
You can create a helper method or extend existing to get headers in the child grid.